dataprofiler/tests/labelers/test_integration_struct_data_labeler.py

```diff
@@ -123,26 +128,18 @@ def test_data_labeler_extend_labels(self):
                                     "label mapping to predict."):
             model_predictions = data_labeler.predict(data=self.df[0])
 
-
-        # get char-level predictions on default model
-        expected_label_mapping = {
-            "PAD": 0, "CITY": 1, "BACKGROUND": 1, "ADDRESS": 2, "BAN": 3,
-            "CREDIT_CARD": 4, "EMAIL_ADDRESS": 5, "UUID": 6, "HASH_OR_KEY": 7,
-            "IPV4": 8, "IPV6": 9, "MAC_ADDRESS": 10, "NAME": 11, "PERSON": 11,
-            "PHONE_NUMBER": 12, "SSN": 13, "URL": 14, "DATETIME": 15,
-            "INTEGER_BIG": 16, "INTEGER": 16, "FLOAT": 17, "QUANTITY": 18,
-            "ORDINAL": 19, "NEW_LABEL": 20}
         model_predictions = data_labeler.fit(x=self.df[0], y=self.df[1])
 
         self.assertEqual(1, len(model_predictions))
         self.assertEqual(3, len(model_predictions[0]))  # history, f1, f1_report
         self.assertIsInstance(model_predictions[0][0], dict)  # history
         self.assertIsInstance(model_predictions[0][1], float)  # f1
         self.assertIsInstance(model_predictions[0][2], dict)  # f1_report
-        self.assertDictEqual(expected_label_mapping, data_labeler.label_mapping)
+        self.assertTrue(new_label in data_labeler.label_mapping) # Ensure new label added
```

Comment:
If we want to keep this assert, i would suggest using:
`self.assertIn`. However, I do think it would be better to assert that the label is at the lend of the label set and that the original labels are unchanged.

One option would be to manually set the labels, then add, because then we control the labels.
Another option would be to grab the label mapping before and after the add. validate it is just an add to the end. however, this no longer becomes explicit and could succumb to a relational error.

These current two tests, while unlikely could still have errors if add_label were altered i.e. the label added coverts all labels to something else but maintains that label.

Just my thoughts.

Reply:
See update